src/utils/codemods/findAvailableCodemodsInNodeModules.js

```diff
@@ -32,45 +49,62 @@ const hasCodemods = codemodsPath =>
  * @returns {String[]}
  */
 const findCodemodsFolders = paths => {
-    return fs
-        .readdirSync(paths.NODE_MODULES)
-        .filter(name => {
-            const curPath = path.join(paths.NODE_MODULES, name)
-
-            if (isDotFolder(name)) return false
-            if (!isFolder(curPath)) return false
+    if (!fs.existsSync(paths.PACKAGE_JSON)) {
+        return []
+    }
 
-            return true
+    let packageJson
+    try {
+        const packageJsonRaw = fs.readFileSync(paths.PACKAGE_JSON, {
+            encoding: 'utf8',
         })
-        .reduce((acc, cur) => {
-            const curPath = path.join(paths.NODE_MODULES, cur)
-
-            // Node module is not namespace and contains codemods
-            if (
-                isFolder(path.join(curPath, 'codemods')) &&
-                hasCodemods(path.join(curPath, 'codemods'))
-            ) {
-                return [...acc, cur]
-            }
-
-            // Node module is a namespace containing node modules that contain codemods
-            const curPathFolders = fs
-                .readdirSync(curPath)
-                .filter(
-                    name =>
-                        isNotDotFolder(name) &&
-                        isFolder(path.join(curPath, name, 'codemods')) &&
-                        hasCodemods(path.join(curPath, name, 'codemods'))
-                )
-                .map(name => `${cur}/${name}`)
-
-            return [...acc, ...curPathFolders]
-        }, [])
+        packageJson = JSON.parse(packageJsonRaw)
+    } catch (e) {
+        log.warn('Could not parse package.json')
+        log.debug(`Path to package.json: ${paths.PACKAGE_JSON}`)
+        return []
+    }
+
+    if (!packageJson.dependencies && !packageJson.devDependencies) {
+        log.debug('No (dev/prod) dependencies found in package.json')
+        log.debug(`Path to package.json: ${paths.PACKAGE_JSON}`)
+        return []
+    }
+
+    const dependencies = [
+        ...Object.keys(packageJson.dependencies || {}),
+        ...Object.keys(packageJson.devDependencies || {}),
+    ]
+
+    return dependencies.reduce((acc, dependency) => {
+        const depPath = getPathOfDependency(dependency)
+
+        if (!depPath) {
+            return acc
+        }
+
+        const codemodPath = path.join(depPath, 'codemods')
+
+        if (!isFolder(codemodPath)) {
+            return acc
+        }
+
+        if (!hasCodemods(codemodPath)) {
+            return acc
+        }
+
+        return [...acc, dependency]
```

Comment:
*minor* it's sightly more performant to mutate the result:

```
acc.push(dependency)
return acc
```

But that's less "functional", shouldn't be a big deal either way here

Reply:
Shouldn't matter in this case